aten/src/ATen/core/TransformationHelper.h

```diff
@@ -74,4 +76,59 @@ C10_HOST_DEVICE inline dist_acctype<T> uniform_real(V val, T from, T to) {
   return (x * (to - from) + from);
 }
 
+/**
+ * Transforms normally distributed `val` with mean 0.0 and standard deviation 1.0 to
+ * normally distributed with `mean` and standard deviation `std`.
+ */
+template <typename T>
+C10_HOST_DEVICE inline T normal(T val, T mean, T std) {
+  return val * std + mean;
+}
+
+/**
+ * Transforms uniformly distributed `val` between 0.0 and 1.0 to
+ * Cauchy distribution with location parameter `median` and scale parameter `sigma`.
+ */
+template <typename T>
+C10_HOST_DEVICE inline T cauchy(T val, T median, T sigma) {
+  // https://en.wikipedia.org/wiki/Cauchy_distribution#Cumulative_distribution_function
+  return median + sigma * at::tan(static_cast<T>(M_PI) * (val - static_cast<T>(0.5)));
+}
+
+/**
+ * Transforms uniformly distributed `val` between 0.0 and 1.0 to
+ * exponentialy distributed with `lambda` parameter of the distribution.
+ */
+template <typename T>
+C10_HOST_DEVICE __ubsan_ignore_float_divide_by_zero__ inline T exponential(T val, T lambda) {
+  // https://en.wikipedia.org/wiki/Exponential_distribution#Generating_exponential_variates
+  // Different implementations for CUDA and CPU to preserve original logic
+  // TODO: must be investigated and unified!!!
```

Comment:
This is old, but can it be unified for `val` in [0,1) --vs-- (0,1] ?
`log(1- val)` for `val` in [0, 1), 1 must be exclusive
`log(val)` for `val` in (0, 1], 0 must be exclusive

Reply:
feel free to send a patch for it